File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n= prices.size();
        int maxProfit=0;
        int minprofit=prices[0];
    for(int buy=1; buy<n; buy++)
    { 
     minprofit=min(prices[buy],minprofit);
    maxProfit=max(maxProfit,prices[buy]-minprofit);
    }
    return maxProfit;
    }
};
 /* for ( int buy=0; buy<n; buy++)
        {
            for ( int sell = buy+1; sell<n; sell++)
            {
                int profit=prices[sell]-prices[buy];
                 maxProfit=max(profit,maxProfit);
                 
            }
        }
    return maxProfit; */

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
     int profit=0;
     int buyPrice=Integer.MAX_VALUE;
     int maxiProfit=Integer.MIN_VALUE;
        for ( int i = 0; i< prices.length;i++)
        { 
            if (buyPrice<prices[i])
            {
                profit=prices[i]-buyPrice;
               maxiProfit=Math.max(maxiProfit,profit);
            }
            else
            {   
                buyPrice=prices[i];
                 maxiProfit=Math.max(maxiProfit,profit);
            }
            }
            return maxiProfit;
        }
    }
